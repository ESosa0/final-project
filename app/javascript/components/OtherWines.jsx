import React from 'react'

const OtherWines = (props) => (
  <div className="other-wines">
    <p>Don't like <span className="titleize">{props.wineName}?</span></p>
    <p><span className="link" onClick={props.getNextWine}> See another wine</span> that goes nicely with your dish.</p>
  </div>
)

export default OtherWines