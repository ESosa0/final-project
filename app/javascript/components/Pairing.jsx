import React from 'react'
import {connect} from 'react-redux'
import WhiteBottle from '../../assets/images/wine_bottle_white.svg'
import PinkBottle from '../../assets/images/wine_bottle_pink.svg'
import RedBottle from '../../assets/images/wine_bottle_red.svg'
import { Link } from 'react-router-dom'
import OtherWines from './OtherWines'

export class Pairing extends React.Component {
  state = {
    currentWine: this.props.wineList[0],
    currentWineIndex: 0
  }

  getWineBottle(wine) {
    if (wine.body < 4){
      return WhiteBottle
    } else if (wine.body > 4){
      return RedBottle
    } else {
      return PinkBottle
    }
  }

  setCurrentWine = () => {
    const currentWineIndex = this.state.currentWineIndex < this.props.wineList.length - 1
      ? this.state.currentWineIndex + 1
      : 0

    this.setState({
      currentWine: this.props.wineList[currentWineIndex], 
      currentWineIndex 
    })
  }

  render(){
    const {wineList, selections, dishName} = this.props 
    const {currentWine} = this.state

    return(
      <section className="box">
        <h3 className="capitalize dish-name">{dishName}</h3>
        <p className="dish-name-subheading">goes well with</p>
        <h1 className="titleize pairing-name">{currentWine.name}</h1>
        <div className="pairing-bottles">
          <img src={this.getWineBottle(currentWine)} className="pairing-bottle"/>
          <img src={this.getWineBottle(currentWine)} className="pairing-bottle"/>
          <img src={this.getWineBottle(currentWine)} className="pairing-bottle"/>
        </div>

        {wineList.length > 1 && 
            <OtherWines 
              wineName={currentWine.name} 
              getNextWine={this.setCurrentWine} 
            />
        }

        <Link to={`/`} className="next-button">Get a new pairing</Link>
      </section>
    ) 
  }
}

const mapStateToProps = state => ({
  wineList: state.wineList,
  selections: state.selections,
  dishName: state.dishName
})

export default connect(mapStateToProps)(Pairing)
