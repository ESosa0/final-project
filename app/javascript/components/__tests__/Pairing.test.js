import {Pairing} from '../Pairing'
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'
import {connect} from 'react-redux'
import OtherWines from '../OtherWines'

describe("Pairing Component", () => {
  it("renders a default pairing component", () => {
    const component = shallow(
      <Pairing
        wineList = {[
          {
            name: "Merlot"
          }
        ]}
        dishName = {"Braised Beef with Mushroom"}
      />
    )
    expect(toJSON(component)).toMatchSnapshot()
  })

  it("renders multiple wines", () => {
    const component = shallow(
      <Pairing
        wineList = {[
          {
            name: "Merlot"
          },
          {
            name: "Pinot Gris"
          }
        ]}
        dishName = {"Braised Beef with Mushroom"}
      />
    )
    const otherWines = component.find(OtherWines)
    expect(toJSON(otherWines)).toMatchSnapshot()
  })
  
  it("renders next wine in a list when getNextWine is called", () => {
    const component = shallow(
      <Pairing
        wineList = {[
          {
            name: "Merlot"
          },
          {
            name: "Pinot Gris"
          }
        ]}
        dishName = {"Braised Beef with Mushroom"}
      />
    )
    const otherWines = component.find(OtherWines)
    otherWines.props().getNextWine()

    component.update()
    setImmediate(() => {
      expect(toJSON(component)).toMatchSnapshot()
    })
  })
})


