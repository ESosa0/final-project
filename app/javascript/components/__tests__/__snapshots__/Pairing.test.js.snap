// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pairing Component renders a default pairing component 1`] = `
<section
  className="box"
>
  <h3
    className="capitalize dish-name"
  >
    Braised Beef with Mushroom
  </h3>
  <p
    className="dish-name-subheading"
  >
    goes well with
  </p>
  <h1
    className="titleize pairing-name"
  >
    Merlot
  </h1>
  <div
    className="pairing-bottles"
  >
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
  </div>
  <Link
    className="next-button"
    replace={false}
    to="/"
  >
    Get a new pairing
  </Link>
</section>
`;

exports[`Pairing Component renders multiple wines 1`] = `
<OtherWines
  getNextWine={[Function]}
  wineName="Merlot"
/>
`;

exports[`Pairing Component renders next wine in a list when getNextWine is called 1`] = `
<section
  className="box"
>
  <h3
    className="capitalize dish-name"
  >
    Braised Beef with Mushroom
  </h3>
  <p
    className="dish-name-subheading"
  >
    goes well with
  </p>
  <h1
    className="titleize pairing-name"
  >
    Merlot
  </h1>
  <div
    className="pairing-bottles"
  >
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
    <img
      className="pairing-bottle"
      src="test-file-stub"
    />
  </div>
  <OtherWines
    getNextWine={[Function]}
    wineName="Merlot"
  />
  <Link
    className="next-button"
    replace={false}
    to="/"
  >
    Get a new pairing
  </Link>
</section>
`;
